src/tests/Room.test.jsx

```diff
@@ -0,0 +1,45 @@
+import { test, describe, expect } from 'vitest';
```

Comment:
- [ ] Although your tests are on point but as the requirement says, you must use React-testing library for integration tests.
- Create integration tests for your application using the React Testing Library.
- You may need to mock the access to the API, so that your tests don't send actual requests.
- You may need to mock the connection to the Redux Store.